akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessKeepAlivePinger.scala

```diff
@@ -0,0 +1,292 @@
+          paused(state.revision - 1)
+        }
+
+      case InternalGetResponse(_ @Replicator.NotFound(`ddataKey`)) =>
+        context.log.debug(
+          "{}: No previous state stored for Sharded Daemon Process, starting from revision [0] with the initial [{}] workers",
+          daemonProcessName,
+          initialNumberOfInstances)
+        subscribeToTopic()
+        startTicks(startRevision)
+        start(startRevision, numberOfProcesses = initialNumberOfInstances)
+
+      case InternalGetResponse(_ @Replicator.GetFailure(`ddataKey`)) =>
+        context.log
+          .debug("{}: Failed fetching initial state for Sharded Daemon Process pinger, retrying", daemonProcessName)
+        init()
+
+    }
+
+  }
+
+  private def start(currentRevision: Long, numberOfProcesses: Int): Behavior[Message] = {
+    val sortedIdentities =
+      ShardedDaemonProcessId.sortedIdentitiesFor(currentRevision, numberOfProcesses, supportsRescale)
+
+    Behaviors.receiveMessage {
+      case Tick(`currentRevision`) =>
+        if (shouldPing()) {
+          context.log.debugN(
+            s"Sending periodic keep alive for Sharded Daemon Process [{}] to [{}] processes (revision [{}]).",
+            daemonProcessName,
+            sortedIdentities.size,
+            currentRevision)
+          context.pipeToSelf(sendKeepAliveMessages(sortedIdentities)) { _ =>
+            SendKeepAliveDone(currentRevision)
+          }
+        } else {
+          timers.startSingleTimer(Tick, Tick(currentRevision), settings.keepAliveInterval)
+        }
+        Behaviors.same
+      case SendKeepAliveDone(`currentRevision`) =>
+        timers.startSingleTimer(Tick, Tick(currentRevision), settings.keepAliveInterval)
+        Behaviors.same
+      case Pause(revision, replyTo) =>
+        context.log.debug2("Pausing sharded daemon process pinger [{}] (revision [{}])", daemonProcessName, revision)
+        if (revision <= currentRevision) {
+          timers.cancel(Tick)
+        }
+        replyTo ! StatusReply.Success(context.self.path)
+        paused(revision)
+
+      case SendKeepAliveDone(oldRevision) =>
+        context.log.debugN(
+          "KeepAlive for [{}] got SendKeepAliveDone from old revision [{}], current is [{}], ignoring",
+          daemonProcessName,
+          oldRevision,
+          currentRevision)
+        Behaviors.ignore
+
+      case Tick(oldRevision) =>
+        context.log.debugN(
+          "Keep alive for [{}] got tick from old revision [{}], current is [{}], ignoring",
+          daemonProcessName,
+          oldRevision,
+          currentRevision)
+        Behaviors.ignore
+      case Resume(`currentRevision`, _, replyTo) =>
+        context.log.debug2(
+          "Sharded daemon process pinger [{}] got start for already started revision (revision [{}]",
+          daemonProcessName,
+          currentRevision)
+        replyTo ! StatusReply.Success(context.self.path)
+        Behaviors.ignore
```

Comment:
doesn't `Behaviors.ignore` mean that it will ignore all incoming messages after this?
```
 A behavior that ignores every incoming message and returns “same”.
 ```
 
 Shouldn't it be `Behaviors.same`?

Reply:
🤦 fixed in 2ab0b18ffdb9d02e4a79b728efa2a2bd9c5b7f20